packages/typescript-estree/src/create-program/getScriptKind.ts

```diff
@@ -0,0 +1,49 @@
+import path from 'path';
+import * as ts from 'typescript';
+
+function getScriptKind(filePath: string, jsx: boolean): ts.ScriptKind {
+  const extension = path.extname(filePath).toLowerCase();
+  // note - we respect the user's extension when it is known we could override it and force it to match their
+  // jsx setting, but that could create weird situations where we throw parse errors when TSC doesn't
```

Comment:
Err, is the grammar off on this one? I'm not parsing it.

Reply:
lmao i left it in because it was there.
I don't know I was probably pretty cooked when I wrote it way back when.

it's just a bad way of saying "we only respect the jsx setting if TS doesn't already have a preference for the extension so we match TS exactly"